csharp/src/ProtocolBuffers.Test/TestProtos/Unittest.cs

```diff
@@ -6377,104 +6417,137 @@ public sealed partial class Builder : pb::GeneratedBuilder<TestAllTypes, Builder
       }
 
       public bool HasOneofUint32 {
-        get { return result.hasOneofUint32; }
+        get { return result.oneofFieldCase_ == OneofFieldOneofCase.OneofUint32; }
       }
       public uint OneofUint32 {
-        get { return result.OneofUint32; }
+        get {
+          if (result.oneofFieldCase_ == OneofFieldOneofCase.OneofUint32) {
+            return (uint) result.oneofField_;
+          } else {
+            return 0;
+          }
+        }
         set { SetOneofUint32(value); }
       }
       public Builder SetOneofUint32(uint value) {
         PrepareBuilder();
-        result.hasOneofUint32 = true;
-        result.oneofUint32_ = value;
+        result.oneofField_ = value;
+        result.oneofFieldCase_ = OneofFieldOneofCase.OneofUint32;
         return this;
       }
       public Builder ClearOneofUint32() {
         PrepareBuilder();
-        result.hasOneofUint32 = false;
-        result.oneofUint32_ = 0;
+        if (result.oneofFieldCase_ == OneofFieldOneofCase.OneofUint32) {
+          result.oneofFieldCase_ = OneofFieldOneofCase.OneofFieldNotSet;
+        }
         return this;
       }
 
       public bool HasOneofNestedMessage {
-       get { return result.hasOneofNestedMessage; }
+       get { return result.oneofFieldCase_ == OneofFieldOneofCase.OneofNestedMessage; }
       }
       public global::Google.ProtocolBuffers.TestProtos.TestAllTypes.Types.NestedMessage OneofNestedMessage {
-        get { return result.OneofNestedMessage; }
+        get { return result.oneofFieldCase_ == OneofFieldOneofCase.OneofNestedMessage ? (global::Google.ProtocolBuffers.TestProtos.TestAllTypes.Types.NestedMessage) result.oneofField_ : global::Google.ProtocolBuffers.TestProtos.TestAllTypes.Types.NestedMessage.DefaultInstance; }
```

Comment:
I'm surprised that this code uses the conditional operator, when earlier code used "if". Both will work, but I'd have expected it to be easier to have consistency in the generator.